File: src/main/java/com/itnoduck/acmate/problem/entity/Problem.java
package com.itnoduck.acmate.problem.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 题目实体，映射 problem 表。
 *
 * <p>tags 当前存储为逗号分隔字符串（VARCHAR(255)），暂不拆分为关联表或 JSON 字段。</p>
 */
@Data
@TableName("problem")
public class Problem {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String platform;
    private String externalProblemKey;
    private String title;
    private String sourceUrl;
    private String difficulty;
    private String tags;
    private String contentMd;
    private Long creatorUserId;
    private Integer status;
    private String deactivationSource;
    private String deactivationReason;
    private Long deactivatedBy;
    private LocalDateTime deactivationTime;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: frontend/src/types/problem.ts
export interface ProblemSummary {
  id: number
  platform: string
  externalProblemKey: string | null
  title: string
  sourceUrl: string | null
  difficulty: string | null
  tags: string | null
  creatorUserId: number
  creatorUsername: string | null
  creatorNickname: string | null
  createTime: string
}

export interface ProblemDetail {
  id: number
  platform: string
  externalProblemKey: string | null
  title: string
  sourceUrl: string | null
  difficulty: string | null
  tags: string | null
  contentMd: string | null
  creatorUserId: number
  creatorUsername: string | null
  creatorNickname: string | null
  active: boolean
  deactivationSource: string | null
  createTime: string
  updateTime: string
}

export interface PageResponse<T> {
  page: number
  size: number
  total: number
  pages: number
  records: T[]
}

export interface ProblemQueryParams {
  page?: number
  size?: number
  platform?: string
  difficulty?: string
  keyword?: string
  creatorUserId?: number
}

export interface CreateProblemRequest {
  platform: string
  externalProblemKey?: string
  title: string
  sourceUrl?: string
  difficulty?: string
  tags?: string
  contentMd?: string
}

export type UpdateProblemRequest = CreateProblemRequest

export type ProblemStatusView = 'ACTIVE' | 'INACTIVE'

export type MineProblemStatusFilter = 'ALL' | 'ACTIVE' | 'INACTIVE'

export interface MyProblemSummary {
  id: number
  platform: string
  externalProblemKey: string | null
  title: string
  sourceUrl: string | null
  difficulty: string | null
  tags: string | null
  status: ProblemStatusView
  createTime: string
  updateTime: string
}

export interface MyProblemQueryParams {
  page?: number
  size?: number
  platform?: string
  difficulty?: string
  keyword?: string
  status?: MineProblemStatusFilter
}

export interface AdminProblemSummary {
  id: number
  platform: string
  externalProblemKey: string | null
  title: string
  sourceUrl: string | null
  difficulty: string | null
  tags: string | null
  status: ProblemStatusView
  deactivationSource: string | null
  deactivationReason: string | null
  deactivatedBy: number | null
  deactivationTime: string | null
  creatorUserId: number
  creatorUsername: string | null
  creatorNickname: string | null
  createTime: string
  updateTime: string
}

export interface AdminProblemQueryParams {
  page?: number
  size?: number
  platform?: string
  difficulty?: string
  keyword?: string
  creatorUserId?: number
  status?: MineProblemStatusFilter
}

export interface AdminDeactivateRequest {
  reason: string
}
